src/analysis/normed_space/pointwise.lean

```diff
@@ -95,13 +95,110 @@ by rw [vadd_ball, vadd_eq_add, add_zero]
 lemma vadd_closed_ball_zero (x : E) (r : ℝ) : x +ᵥ closed_ball 0 r = closed_ball x r :=
 by rw [vadd_closed_ball, vadd_eq_add, add_zero]
 
-variables [normed_space ℝ E]
+variables [normed_space ℝ E] {x y z : E} {δ ε : ℝ}
 
 /-- In a real normed space, the image of the unit ball under scalar multiplication by a positive
 constant `r` is the ball of radius `r`. -/
 lemma smul_unit_ball_of_pos {r : ℝ} (hr : 0 < r) : r • ball 0 1 = ball (0 : E) r :=
 by rw [smul_unit_ball hr.ne', real.norm_of_nonneg hr.le]
 
+-- This is also true for `ℚ`-normed spaces
```

Comment:
What exactly do you need in order to generalize this?

Reply:
Nothing. Everything is here (namely `exists_rat_btwn`). What I'm afraid of is that writing those lemmas as `normed_space ℚ E` won't make them directly applicable to `normed_space ℝ E`. Maybe I'm wrong?

Reply:
Okay, it's indeed not automatic, but could be made so if we were able to ensure everything is defeq for char zero division rings when adding the `normed_space ℝ E → normed_space ℚ E` instance (similar to [`normed_space.complex_to_real`](https://leanprover-community.github.io/mathlib_docs/analysis/complex/basic.html#normed_space.complex_to_real)).

So what about we just leave it as is for now and worry about it when we will actually deal with `ℚ`-normed spaces?

Reply:
I was wondering whether `ℝ` could be replaced by a suitable normed field `K`. For some definition of "suitable".

Reply:
But I'm happy to leave it as it is for now.

Reply:
I think the answer is `linear_ordered_nondiscrete_normed_field`, which doesn't currently exist.